src/persist-types/src/columnar.rs

```diff
@@ -125,9 +125,16 @@ pub trait ColumnPush<T: Data>: ColumnMut<T::Cfg> {
     fn push<'a>(&mut self, val: T::Ref<'a>);
 }
 
+/// A common trait implemented by `Data::Mut` types. Provides the ability to
+/// "finish" a mutable array, returning the corresponding `Data::Col`.
+pub trait ColumnFinish<T: Data>: ColumnMut<T::Cfg> {
```

Comment:
It seems like `ColumnFinish` is implemented for exactly the same types as `ColumnPush` is. (Which makes some intuitive sense to me as well... both should be implemented for the builder types.) Can we combine them into a single trait?

Reply:
That makes sense! I'll add `finish(...)` to `ColumnPush`